resolve/index.go

```diff
@@ -61,13 +61,23 @@ type Resolver interface {
 	Resolve(c *config.Config, ix *RuleIndex, rc *repo.RemoteCache, r *rule.Rule, imports interface{}, from label.Label)
 }
 
+// CrossResolver is an interface that language extensions can implement to provide
+// custom dependency resolution logic for other languages.
+type CrossResolver interface {
+	// CrossResolve attempts to resolve an import string to a rule for languages
+	// other than the implementing extension. lang is the langauge of the rule
+	// with the dependency.
+	CrossResolve(c *config.Config, ix *RuleIndex, rc *repo.RemoteCache, imp ImportSpec, lang string) []FindResult
```

Comment:
Let's drop `rc` from here. It seems to require a `*repo.RemoteCache` in a lot of places without really providing value.

I was thinking of generalizing `repo.RemoteCache` (it's very Go-specific right now), but now I think it would be better for extensions to just keep a cache like that themselves instead of passing around a shared cache.